subprojects/common/http-client/src/main/kotlin/com/avito/http/HttpClientProvider.kt

```diff
@@ -0,0 +1,72 @@
+package com.avito.http
+
+import com.avito.android.stats.CountMetric
+import com.avito.android.stats.SeriesName
+import com.avito.android.stats.StatsDSender
+import com.avito.http.TryFailCallback.Companion.combine
+import com.avito.http.internal.ServiceMetricsInterceptor
+import com.avito.http.internal.StatsdServiceEventsListener
+import com.avito.logger.LoggerFactory
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import java.util.concurrent.TimeUnit
+
+public class HttpClientProvider(
+    private val statsDSender: StatsDSender,
+    private val loggerFactory: LoggerFactory,
+    private val builderTransform: OkHttpClient.Builder.() -> OkHttpClient.Builder = { this }
+) {
+
+    private val builder = OkHttpClient.Builder()
+
+    public fun provide(
+        serviceName: String,
```

Comment:
Why don't we take it from the request? 
A client can lie (make a mistake), but a request can't.

Reply:
Did you think how to transform request to meaningful series already?

Reply:
I thought about simple mapping for REST:

`http://integration.api/storage/files/upload` -> `integration_api.storage_files_upload`

And the ability to override method description for JSON RPC:

`http://integration.api/action` -> `integration_api.<request descriptor>`

"request descriptor" is calculated in advance and passed by tag: https://github.com/avito-tech/avito-android/pull/841#discussion_r594924626

Some additional considerations.

Why do we need "serviceName" at all? I see only one reason, to make it stable against changes in the host.
`http://integration.api` -->`http://proxy-integration.api`
We can live without it. 
A new endpoint means a different route. I'd like to distinguish metrics.

Reply:
we have api's with schemes like: `http://prod.k/service-name/api/methods` and `http://service.name/api/methods`
i don't know how to group this properly

Reply:
Fixed